pages/slice/language-reference/slice-grammar.md

```diff
@@ -2,14 +2,587 @@
+
+An operation consists of an [identifier], followed be a list of [parameters][parameter], optionally followed by a return type, optionally followed by an [exception specification][exception-specification-guide].
+Operations support the `idempotent` modifier keyword. For information about its effects, see the [idempotent][idempotent-guide] section.
+Additionally, [local attributes][attribute] and a [doc-comment] may be applied to the operation in its prelude.
+
+```ebnf {% showTitle=false %}
+Operation
+    : Prelude "idempotent"? Identifier "(" UndelimitedList<Parameter> ")" ReturnType? ExceptionSpecification?
+    ;
+```
+
+There are two syntaxes for specifying a return type, depending on the number of elements you're returning.
+A single return type can be specified by an arrow, followed by an optional [tag], followed by an optional [`stream`][stream-guide] keyword, and then a [type].
+The syntax of a single return type is identical to a [parameter], but without an identifier or colon separator.
+Multiple return parameters can be specified as a tuple. This consists of an arrow, followed by a list of [parameters][parameter] wrapped in parenthesis.
+It is illegal for a return tuple to contain less than 2 elements. 
+
+```ebnf {% showTitle=false %}
+ReturnType
+    : "->" Tag? "stream"? TypeRef
+    | "->" "(" UndelimitedList<Parameter> ")"
+    ;
+```
+Parameter lists may be optionally separated by commas. This applies to both operation parameters and return parameters.
+
+There are two syntaxes for [exception specifications][exception-specification-guide], depending on the number of [exceptions][exception] being specified.
+A single exception can be specified with the `throws` keyword, followed by the (possibly scoped) identifier of an exception.
+Multiple exceptions can be specified as a tuple. This consists of the `throws` keyword, followed by the (possibly scoped) identifiers of one or more exceptions, separated by commas.
+Exception specifications can only declared in [`Slice1`][compilation-mode-guide] mode.
+
+```ebnf {% showTitle=false %}
+ExceptionSpecification
+    : "throws" TypeRef
+    | "throws" "(" NonEmptyCommaList<TypeRef> ")"
+    ;
+```
+
+For additional information on operations, see the [operation][operation-guide] pages.
+
+### Parameters
+
+The syntax for parameters is identical to the syntax for [fields][field], but with added support for the `stream` modifier keyword.
+
+Parameters consist of an [identifier], followed by a colon, and then a [type].
+Optionally, a [tag] may be applied to the parameter, directly before its identifier, making this a [tagged parameter][tag].
+Additionally, [local attributes][attribute] and a [doc-comment] may be applied to the field in its prelude.
+
+```ebnf {% showTitle=false %}
+Parameter
+    : Prelude Tag? Identifier ":" "stream"? TypeRef
+    ;
+```
+
+Parameters also support the [`stream`][streamed-parameters-guide] modifier keyword, which may be applied to the parameter's type.
```

Comment:
> Parameters also support the [stream][streamed-parameters-guide] modifier keyword,

It seems redundant to mention the support for the 'stream' modifier keyword here since it's already stated in the previous paragraph.

Reply:
I simplified the first sentence to just say `..., but with added support for streams`
Instead of re-using the exact same terminology we use down here when we explain it in greater detail.